### PR TITLE
Adapt to deprecation of spurt in upstream Mojolicious

### PR DESCRIPTION
**Pull request by okurz**
Related progress issue: https://progress.opensuse.org/issues/135632

### CODE REVIEW

File: t/26-controllerrunning.t
Comment (perlpunk) [inline t/26-controllerrunning.t:73]: That's wrong usage now, I think it needs to be
```suggestion
    $t_file->spew(join '', @fake_data);
```
```
sub spew {
  my ($self, $content, $encoding) = @_;
...
}
  
sub spurt {
  deprecated 'Mojo::File::spurt is deprecated in favor of Mojo::File::spew';
  shift->spew(join '', @_);
}
```

File: t/config.t
Comment (perlpunk) [inline t/config.t:220]: Same here

**Comment by perlpunk** [comment]
I think this has to be on top of https://github.com/os-autoinst/openQA/pull/5304 to pass.

**Comment by okurz** [comment]
I suggest to have https://github.com/os-autoinst/os-autoinst-common/pull/27 merged first and then pull in and rebase here

File: t/26-controllerrunning.t
Comment (Martchus) [reply t/26-controllerrunning.t:73]: Or can you pass an array ref?

Comment (perlpunk) [reply t/26-controllerrunning.t:73]: Nope: https://metacpan.org/pod/Mojo::File#spew
```
$path = $path->spew($bytes);
$path = $path->spew($chars, 'UTF-8');
```

Comment (okurz) [reply t/26-controllerrunning.t]: done

File: t/config.t
Comment (okurz) [reply t/config.t]: done

File: t/api/04-jobs.t
Comment (Martchus) [reply t/api/04-jobs.t:47]: Yes, looks like it.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: t/api/04-jobs.t
Comment (okurz) [reply t/api/04-jobs.t]: done. Looked up all occurences with `spew.*@`

**Review by Martchus — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

File: tools/ci/ci-packages.txt
Comment (perlpunk) [inline tools/ci/ci-packages.txt:200]: We also need:
* perl-DBD-Pg-3.16.3
* perl-Perl-Tidy-20230701

**Review by baierjan — APPROVED** [review]

**Comment by okurz** [comment]
waiting for https://build.opensuse.org/package/live_build_log/devel:openQA:ci/base/containers/x86_64 first.

**Comment by okurz** [comment]
I don't understand why https://app.circleci.com/pipelines/github/os-autoinst/openQA/12196/workflows/37018546-f44b-4938-b133-98bbd86c4087/jobs/113854 fails with

```
Package 'perl-Perl-Tidy-20230909' not found.
```

From https://app.circleci.com/pipelines/github/os-autoinst/openQA/12196/workflows/37018546-f44b-4938-b133-98bbd86c4087/jobs/113854 I understand that the container image is registry.opensuse.org/devel/openqa/ci/containers/base:latest and in https://app.circleci.com/pipelines/github/os-autoinst/openQA/12196/workflows/37018546-f44b-4938-b133-98bbd86c4087/jobs/113854?invite=true#step-108-9_106 repositories https://download.opensuse.org/repositories/devel:/openQA:/Leap:/15.5/15.5 and http://download.opensuse.org/repositories/devel:openQA/15.5 are added which should have the package from https://build.opensuse.org/package/show/devel:openQA:Leap:15.5/perl-Perl-Tidy

**Comment by okurz** [comment]
oh no.

```
1: Mojo::File::spurt is deprecated in favor of Mojo::File::spew at /usr/lib/perl5/vendor_perl/5.26.1/Mojolicious/Plugin/AssetPack/Store.pm line 142.
```

**Comment by okurz** [comment]
Blocked by https://github.com/mojolicious/mojo-assetpack/pull/149

**Comment by perlpunk** [comment]
> Blocked by [mojolicious/mojo-assetpack#149](https://github.com/mojolicious/mojo-assetpack/pull/149)

I can add a patch in OBS later today to speed this up

**Comment by perlpunk** [comment]
> > Blocked by [mojolicious/mojo-assetpack#149](https://github.com/mojolicious/mojo-assetpack/pull/149)
> 
> I can add a patch in OBS later today to speed this up

https://build.opensuse.org/request/show/1112340

**Comment by perlpunk** [comment]
@okurz It is usually a good idea to not modify meta files like Changes or the version number of modules in a pull request. That is something the maintainer wants to do themselves.
I had to modify your patch so that it applied cleanly on what's currently in OBS. For example you changed the version number from 2.15 to 2.16, but in OBS we are still at 2.14, and I believe 2.15 would already be the next version number.

**Comment by perlpunk** [comment]
https://build.opensuse.org/request/show/1112340

request was accepted

**Comment by okurz** [comment]
Nice. I guess we need to wait another 1-2 days for this to become effective in the official Tumbleweed repos so same as for ReadWriteProcess I linked packages into devel:openQA. After the build settled there we can retrigger tests here.

**Comment by okurz** [comment]
/usr/lib/perl5/vendor_perl/5.26.1/Mojolicious/Plugin/AssetPack/Store.pm is that part of mojo-assetpack or something different?

**Comment by okurz** [comment]
CI jobs still pull in the old version of assetpack. Within devel:openQA we have a newer build but I am not sure which repo has precedence here. An explicit new version 2.15 would likely help or at least make it explicit from where we would install that dependency

**Comment by perlpunk** [comment]
I think we just need to wait for a new version here: https://download.opensuse.org/tumbleweed/repo/oss/noarch/
Currently I can see the version 2.14-1.3 there.
Interestingly here
https://software.opensuse.org//download.html?project=openSUSE%3AFactory&package=perl-Mojolicious-Plugin-AssetPack
clicking on "Grab binary packages directly" already gives a link to [2.14-2.1](https://download.opensuse.org/repositories/openSUSE:/Factory/standard/noarch/perl-Mojolicious-Plugin-AssetPack-2.14-2.1.noarch.rpm) but it returns a 404.
But as soon as it's there the new version will be higher, so no need for a new module version.

**Pull request merged**
